File: CS/BindingListForServerMode/MainForm.cs
using DevExpress.XtraEditors;
using DevExpress.XtraGrid.Views.Base;
using ServerModeGridProjects;

namespace BindingListForServerMode {
    public partial class MainForm :XtraForm {
        public MainForm() {
            InitializeComponent();
        }

        private void GridView_FocusedRowObjectChanged(object sender, FocusedRowObjectChangedEventArgs e) {
            BindingSource.DataSource = Session.GetLoadedObjectByKey<ServerSideGridTest>(e.Row);
        }
    }
}

File: VB/BindingListForServerMode/MainForm.vb
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGrid.Views.Base
Imports ServerModeGridProjects

Namespace BindingListForServerMode
	Partial Public Class MainForm
		Inherits XtraForm

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub GridView_FocusedRowObjectChanged(ByVal sender As Object, ByVal e As FocusedRowObjectChangedEventArgs) Handles GridView.FocusedRowObjectChanged
			BindingSource.DataSource = Session.GetLoadedObjectByKey(Of ServerSideGridTest)(e.Row)
		End Sub
	End Class
End Namespace